lib/streamlit/elements/alert.py

```diff
@@ -12,86 +12,147 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import cast, TYPE_CHECKING
+from typing import cast, Optional, TYPE_CHECKING
 
 from streamlit.proto.Alert_pb2 import Alert as AlertProto
+from streamlit.errors import StreamlitAPIException
+from streamlit.string_util import is_emoji_valid
 from .utils import clean_text
+import re
 
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
     from streamlit.type_util import SupportsStr
 
+# Function to check the icon parameter on the alert.
+# We check if what's been added is a valid emoji,
+# and default to an empty string if not.
+def check_emoji(emoji):
+    # If the user didn't add an emoji, carry on without it
+    if emoji == None:
+        return clean_text(str(""))
+
+    # Check if the provided character is a valid emoji
+    extracted_emoji = is_emoji_valid(emoji)
+
+    if extracted_emoji is not None:
+        return extracted_emoji
```

Comment:
Hm, I don't think that this function does much more than the `is_emoji_valid` function, so I would prefer to remove this function entirely and call `is_emoji_valid` directly instead within the `AlertMixin` class. The `None`-check behavior currently done here can be moved into the other function.

I also have some small suggestions on the structure of `is_emoji_valid` which I'll add in a comment for that function.

Reply:
Makes sense! Updated on [4390be7](https://github.com/streamlit/streamlit/pull/4964/commits/4390be7616245c5122e9f0c78172c47bdf268aeb)